### PR TITLE
Add a simple select box that allows selecting the syntax highlighter language

### PR DESCRIPTION
**Pull request by ianmcorvidae**
Curious at this point mostly about UX -- is this a thing we want, is this how we want to display it?

Screenshot of current state:

Screenshot_2021-02-09 Discovery Environment

### CODE REVIEW

File: src/components/models/MimeTypes.js [hunk 1]
```diff
@@ -2,14 +2,17 @@ const mimeTypes = {
     PNG: "png",
     GIF: "gif",
     JPEG: "jpeg",
+
```
Comment (ianmcorvidae) [inline src/components/models/MimeTypes.js:5]: heh, didn't mean to commit these. it's fine

**Comment by ianmcorvidae** [comment]
Pulled this out of WIP, ready for review, especially of the UX.

**Comment by ianmcorvidae** [comment]
before and after bundle analysis screenshots, most changes are due to pulling in the async version of react-syntax-highlighter.

syntax-highlight-mode-before
syntax-highlight-mode-after

**Review by sri2k1us — COMMENTED** [review]
Code LGTM. But in term of UX, I like it better if we can put the `Mode` select drop down in the toolbar with other buttons. There is also real estate challenge with that.

**Comment by ianmcorvidae** [comment]
I think I agree. I'll look into/think about how to do that. I might see if an autocomplete works nicely too, since there are a lot of options.

**Review by ash3rz — APPROVED** [review]
Code LGTM too 🎉 Small change adding a lot of options

For UX, I agree it'd be nice to keep the select/autocomplete in the same toolbar.  Perhaps for large screens, it could remain in the toolbar, but for smaller screens it could hide in the dot menu, similar to how the `App Type` filter hides in the dot menu for apps.  It becomes an option that opens a dialog that displays the select/autocomplete.  Just one idea.

**Review by psarando — COMMENTED** [review]
LGTM 👍 
Though I agree with others that it might be better to move the select into the toolbar/dot-menu.

File: src/components/data/viewers/Toolbar.js [hunk 2]
```diff
@@ -138,6 +138,24 @@ function ViewerToolbar(props) {
...
+                disableElevation
+                color="primary"
+                onClick={onClick}
+                startIcon={startIcon}
+                {...props}
+            >
+                <Hidden xsDown>{text}</Hidden>
+            </Button>
+        );
+    };
```
[the remark this answers is not in the document]
Comment (ianmcorvidae) [reply src/components/data/viewers/Toolbar.js:158]: I like this idea. I'd been explicit about passing along the props that were reasonably 'required', but that's a bit much for an inline component in another file :laughing:

**Comment by ianmcorvidae** [comment]
I switched this to an Autocomplete. Screenshots:

Screenshot_2021-03-11 Discovery Environment(2)
Screenshot_2021-03-11 Discovery Environment(1)
Screenshot_2021-03-11 Discovery Environment

**Comment by ianmcorvidae** [comment]
Update: fixed the lowercase "done" to a proper "Done" that's in the one screenshot. Didn't make a new screenshot though.

**Comment by ianmcorvidae** [comment]
Okay, one last fix, set a width on the dropdown that matches the apps filter. It seems to look fine at any width, to me, with that setting.

**Review by ash3rz — APPROVED** [review]
:shipit:

**Comment by ianmcorvidae** [comment]
Cool, thanks for the re-review. I'll merge this now.

**Pull request merged**